### ISSUE TITLE
URL decoder does not decode percent-encoded characters

### ISSUE THREAD

**Issue by TheMarex** [Project-OSRM/osrm-backend#3251]
Our `URLParser` does not actually first decoded `%HEX` encoded characters before doing the real parsing. This for example breaks polylines like in this request:

```
https://router.project-osrm.org/match/v1/car/polyline(%7DjmwFz~ubMyCa@%60@yDJqE)?annotations=false&geometries=geojson&overview=simplified&radiuses=10;10;10;10&steps=true
```

(notice `%7D` and `%60`).

Second part of https://github.com/Project-OSRM/osrm-backend/issues/3037

**Comment by daniel-j-h** [comment]
Isn't this a matter of bringing in node-osrm and then switching to a proper dev HTTPD?

Or do you want to implement a fully fledged [url-encoding](https://en.wikipedia.org/wiki/Percent-encoding) decoder in C++ which will be ripped out in the coming month anyway?

**Comment by TheMarex** [comment]
Maybe we can just bundle something like https://github.com/larroy/urlcpp which should get us there quite fast.

**Comment by TheMarex** [comment]
Addressed by #3252. Thanks @oxidase
